server/routers/orderAPI.ts

```diff
@@ -24,8 +32,26 @@ router.post("/api/order/:id", async (req, res) => {
 
 // Update a specific order
 router.put("/api/order/:id", async (req, res) => {
-  // To be added
-  console.log("Update a specific order");
+  
+  const orderId = req.params["id"];
+  const orderJson = req.body;
+  console.log(`PUT /api/order/${orderId} - ${JSON.stringify(orderJson)}`);
+
+  try {
+    const updatedOrder = await Order.findByIdAndUpdate(orderId, {
+      customer: orderJson.customer,
+      donuts: orderJson.donuts,
+      location: orderJson.location,
+      orderTime: orderJson.orderTime,
+      drone: orderJson.drone,
+      status: orderJson.status,
+    }, { new: true });
+    console.log(`Order with ID ${orderId} updated: ${JSON.stringify(updatedOrder)}`);
```

Comment:
here I can tell that updatedOrder does not reflect orderJson. updatedOrder should reflect orderJson

Reply:
Hmm I don't think this is a problem since the return value of Order.findByIdAndUpdate returns a OrderDocument instead of a JSON.